components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountDisableHandler.java

```diff
@@ -386,6 +408,48 @@ private void setUserClaim(String claimName, String claimValue, UserStoreManager
         }
     }
 
+    private void publishPreAccountDisabledEvent(String accountDisabledEventName, Map<String, Object> map) throws
+            AccountLockException {
+
+        AccountUtil.publishEvent(accountDisabledEventName, AccountUtil.cloneMap(map));
+    }
+
+    private void publishPostAccountDisabledEvent(String accountDisabledEventName, Map<String, Object> map, boolean
+            isDisblePropertySuccessfullyModified) throws AccountLockException {
+
+        Map<String, Object> eventProperties = AccountUtil.cloneMap(map);
+        eventProperties.put(IdentityEventConstants.EventProperty.UPDATED_DISABLED_STATUS,
+                isDisblePropertySuccessfullyModified);
+        AccountUtil.publishEvent(accountDisabledEventName, eventProperties);
+    }
+
+    private static void createAuditMessage(String action, String target, JSONObject dataObject, String result) {
+
+        String loggedInUser = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isBlank(loggedInUser)) {
+            loggedInUser = AuditConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        loggedInUser = UserCoreUtil.addTenantDomainToEntry(loggedInUser, tenantDomain);
+        CarbonConstants.AUDIT_LOG.info(String.format(AuditConstants.AUDIT_MESSAGE, loggedInUser, action, target, dataObject, result));
+    }
+
+    private void auditAccountDisable(String action, String target, String userStoreDomainName, String errorMsg,
+                                     String result, Boolean isModifiedStatus) {
+
+        JSONObject dataObject = new JSONObject();
+        dataObject.put(AuditConstants.REMOTE_ADDRESS_KEY, MDC.get(AuditConstants.REMOTE_ADDRESS_QUERY_KEY));
+        dataObject.put(AuditConstants.USER_AGENT_KEY, MDC.get(AuditConstants.USER_AGENT_QUERY_KEY));
+        dataObject.put(AuditConstants.SERVICE_PROVIDER_KEY, MDC.get(AuditConstants.SERVICE_PROVIDER_QUERY_KEY));
+        dataObject.put(AuditConstants.USER_STORE_DOMAIN, userStoreDomainName);
+        dataObject.put(AuditConstants.IS_MODIFIED_STATUS, isModifiedStatus);
+
+        if (AuditConstants.AUDIT_FAILED.equals(result)) {
+            dataObject.put(AuditConstants.ERROR_MESSAGE_KEY, errorMsg);
+        }
+        createAuditMessage(action, target, dataObject, result);
```

Comment:
The same logic is there in both AccountDisableHandler and AccountLockHandler. Therefore let's move this logic to [1] as a new util method and call it here to avoid code repetition.

[1] https://github.com/wso2-extensions/identity-event-handler-account-lock/blob/master/components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/util/AccountUtil.java